### PR TITLE
Cancel job groups in the UI

### LINKED ISSUE #69: Add job cancelation ability to the UI
@chefsalim commented on [Fri Nov 03 2017](https://github.com/habitat-sh/habitat/issues/3998)

Now that we have an API (as well as hab cli cancel command), we should hook up the same functionality as part of the UI.

### UI Prototype
[Job cancel UI Mockup](https://chef.invisionapp.com/share/AREOTA4KQ#/266390910_package_Detail_V2_-_Versions_ALT_Job_Cancel)

#### Notes for prototoype
- :wip:

---

@cnunciato commented on [Fri Nov 03 2017](https://github.com/habitat-sh/habitat/issues/3998#issuecomment-341788307)

cc @ryankeairns

### PR DESCRIPTION
**Pull request by cnunciato**
This change adds support for canceling build job groups in the UI. It also fixes a few minor display bugs related to that functionality, adds a couple of details to the build-detail view (e.g, the resulting package and its channels), fixes an API regression by restoring the JSON bodies of 404 errors (which are currently expected by the UI), and adds an owner_id property to the payload of the job API to allow for navigating from a job to its parent group.

Fixes #69.

Signed-off-by: Christian Nunciato <chris@nunciato.org>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-web/app/shared/_shared.module.scss [hunk 1]
```diff
@@ -3,8 +3,11 @@
 @import 'channels/channels.component';
 @import 'checking-input/checking-input.component';
 @import "copyable/copyable.component";
-@import "docker-export-settings/docker-export-settings.component";
+@import 'docker-export-settings/docker-export-settings.component';
```
Comment (elliott-davis) [inline components/builder-web/app/shared/_shared.module.scss]: Is it supposed to be single or double? The one above is also double

File: components/builder-web/app/shared/job-status-icon/job-status-icon.component.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    let s = '';
+
+    if (this.status) {
+      s = this.status;
+    } else if (this.job) {
+      s = this.job.state || '';
+    }
+
+    return s;
+  }
```
Comment (mgamini) [inline components/builder-web/app/shared/job-status-icon/job-status-icon.component.ts]: could this be `return this.status || this.job ? this.job.state : '';` ?

File: components/builder-web/app/shared/_shared.module.scss (hunk 1 above)
Comment (cnunciato) [reply components/builder-web/app/shared/_shared.module.scss]: Ah, missed one, thanks. (I like single, generally.)

File: components/builder-web/app/shared/job-status-icon/job-status-icon.component.ts (hunk 2 above)
Comment (cnunciato) [reply components/builder-web/app/shared/job-status-icon/job-status-icon.component.ts]: Totally could, yeah.

**Comment by cnunciato** [comment]
Just noticed the button placement and layout aren't quite to design spec. I'll adjust.

**Pull request merged**